core/foundation/inc/ROOT/RMakeUnique.hxx

```diff
@@ -14,20 +14,48 @@
 #ifndef ROOT_RMakeUnique
 #define ROOT_RMakeUnique
 
-#include <memory>
-
 #if __cplusplus < 201402L && !defined(_MSC_VER)
 
+#include <cstddef>
+#include <memory>
+#include <type_traits>
 #include <utility>
 
+// Implementation taken from https://isocpp.org/files/papers/N3656.txt
```

Comment:
Does this violate the licence?

Reply:
@amadio

Reply:
I don't know, there is no license in the link. We should probably stay away just in case. Maybe @Axel-Naumann will know if we can get the code from there.